tests/storage/test_event_federation.py

```diff
@@ -506,11 +508,24 @@ def insert_event(txn):
         )
         self.assertSetEqual(difference, set())
 
-    def test_prune_inbound_federation_queue(self):
-        "Test that pruning of inbound federation queues work"
+    @parameterized.expand(
```

Comment:
We could additionally have this run for all known RoomVersions...

Reply:
Yeah I think it might be better to programmatically build this list out of `KNOWN_ROOM_VERSIONS`, that way we're sure we don't accidentally reproduce this bug in a future room version.